src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -10,12 +11,984 @@ namespace System.Buffers.Text.Tests
+        {
+            int textLength = s_UInt32TextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_UInt32TextArray[i]);
+            }
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData("abcdef")] // standard parse
+        [InlineData("ffffffff")] // max value
+        [InlineData("0")] // min value
+        private static unsafe void ByteSpanToUInt32Hex_BytesConsumed(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            var utf8ByteSpan = new ReadOnlySpan<byte>(utf8ByteArray);
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed, 'X');
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        private static unsafe void ByteSpanToUInt32Hex_BytesConsumed_VariableLength()
+        {
+            int textLength = s_UInt32TextArrayHex.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_UInt32TextArrayHex[i]);
+            }
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed, 'X');
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        #region Inline data
+        [InlineData("107")] // standard parse
```

Comment:
Given the discussion here https://github.com/dotnet/corefxlab/issues/2322#issuecomment-405409110, I wouldn't add all these tests. I recommend only picking a few key ones.

> For example, I would add the single digit case like "0" and at least a special case like leading zeroes ("00000000000000000000123abcdfg"). Here is an example where we have made optimizations for single digit inputs - https://github.com/dotnet/coreclr/pull/18383. We don't need to add the '+' prefixed numbers or all digit length inputs

As an example, for this test, I would keep:
0, min value, max value, "00000000000000000000123abcdfg", "14abcdefghijklmnop", and "-14abcdefghijklmnop".

cc @adamsitnik